controllers/webhook/webhook.go

```diff
@@ -128,42 +129,26 @@ func (w *OpenDataHubValidatingWebhook) Handle(ctx context.Context, req admission
 //+kubebuilder:webhook:path=/mutate-opendatahub-io-v1,mutating=true,failurePolicy=fail,sideEffects=None,groups=datasciencecluster.opendatahub.io,resources=datascienceclusters,verbs=create;update,versions=v1,name=mutate.operator.opendatahub.io,admissionReviewVersions=v1
 //nolint:lll
 
-// OpenDataHubMutatingWebhook is a mutating webhook
 // It currently only sets defaults for modelregiestry in datascienceclusters.
-type OpenDataHubMutatingWebhook struct {
-	Client  client.Client
-	Decoder *admission.Decoder
-}
+type DSCDefaulter struct{}
 
-func (w *OpenDataHubMutatingWebhook) SetupWithManager(mgr ctrl.Manager) {
-	hookServer := mgr.GetWebhookServer()
-	odhWebhook := &webhook.Admission{
-		Handler: w,
-	}
-	hookServer.Register("/mutate-opendatahub-io-v1", odhWebhook)
-}
+// just assert that DSCDefaulter implements webhook.CustomDefaulter.
+var _ webhook.CustomDefaulter = &DSCDefaulter{}
 
-func (w *OpenDataHubMutatingWebhook) Handle(ctx context.Context, req admission.Request) admission.Response {
-	var resp admission.Response
-	resp.Allowed = true // initialize Allowed to be true in case Operation falls into "default" case
-
-	dsc := &dscv1.DataScienceCluster{}
-	err := w.Decoder.Decode(req, dsc)
-	if err != nil {
-		return admission.Denied(fmt.Sprintf("failed to decode request body: %s", err))
-	}
-	resp = w.setDSCDefaults(ctx, dsc)
-	return resp
+func (m *DSCDefaulter) SetupMutateWebhookWithManager(mgr ctrl.Manager) {
+	mutateWebhook := admission.WithCustomDefaulter(mgr.GetScheme(), &dscv1.DataScienceCluster{}, m)
+	mgr.GetWebhookServer().Register("/mutate-opendatahub-io-v1", mutateWebhook)
 }
 
-func (w *OpenDataHubMutatingWebhook) setDSCDefaults(_ context.Context, dsc *dscv1.DataScienceCluster) admission.Response {
+// implement admission.CustomDefaulter interface.
+func (m *DSCDefaulter) Default(_ context.Context, obj runtime.Object) error {
```

Comment:
As this is now extracted to a simple function, how about providing some unit tests for it? It does not even have to employ EnvTest, plain unit testing should be fine here.

Reply:
did some updates on the unit-test.
only for the case which mutate webhook is related.